x-pack/plugins/security_solution/server/endpoint/routes/actions/audit_log_handler.ts

```diff
@@ -5,55 +5,34 @@
  * 2.0.
  */
 
-import { TypeOf } from '@kbn/config-schema';
 import { RequestHandler } from 'kibana/server';
-import { AGENT_ACTIONS_INDEX } from '../../../../../fleet/common';
-import { EndpointActionLogRequestSchema } from '../../../../common/endpoint/schema/actions';
-
+import {
+  EndpointActionLogRequestParams,
+  EndpointActionLogRequestQuery,
+} from '../../../../common/endpoint/schema/actions';
+import { getAuditLogResponse } from './service';
 import { SecuritySolutionRequestHandlerContext } from '../../../types';
 import { EndpointAppContext } from '../../types';
 
 export const actionsLogRequestHandler = (
   endpointContext: EndpointAppContext
 ): RequestHandler<
-  TypeOf<typeof EndpointActionLogRequestSchema.params>,
-  unknown,
+  EndpointActionLogRequestParams,
+  EndpointActionLogRequestQuery,
   unknown,
   SecuritySolutionRequestHandlerContext
 > => {
   const logger = endpointContext.logFactory.get('audit_log');
+
   return async (context, req, res) => {
-    const esClient = context.core.elasticsearch.client.asCurrentUser;
-    let result;
-    try {
-      result = await esClient.search({
-        index: AGENT_ACTIONS_INDEX,
-        body: {
-          query: {
-            match: {
-              agents: req.params.agent_id,
-            },
-          },
-          sort: [
-            {
-              '@timestamp': {
-                order: 'desc',
-              },
-            },
-          ],
-        },
-      });
-    } catch (error) {
-      logger.error(error);
-      throw error;
-    }
-    if (result?.statusCode !== 200) {
-      logger.error(`Error fetching actions log for agent_id ${req.params.agent_id}`);
-      throw new Error(`Error fetching actions log for agent_id ${req.params.agent_id}`);
-    }
+    const {
+      params: { agent_id: elasticAgentId },
+      query: { page = 1, page_size: pageSize = 50 },
```

Comment:
I don't think you should be assigning defaults here. They would go into the schema.
Also - for the API, I would not suggest having the `pageSize` set to 50. I think the normal number I see is 10.

Reply:
done 1db7cbe37cf48b7e91808f316c2a5fc801478b3d. I also learned that we shouldn't be using `schema.maybe` if a nested type within is using a `defaultValue`

https://github.com/elastic/kibana/tree/master/packages/kbn-config-schema#schemamaybe